src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilderInstantiation.cs

```diff
@@ -253,7 +279,72 @@ public override bool ContainsGenericParameters
 
         [RequiresUnreferencedCode("If some of the generic arguments are annotated (either with DynamicallyAccessedMembersAttribute, or generic constraints), trimming can't validate that the requirements of those annotations are met.")]
         public override Type MakeGenericType(params Type[] inst) { throw new InvalidOperationException(SR.Format(SR.Arg_NotGenericTypeDefinition, this)); }
-        public override bool IsAssignableFrom([NotNullWhen(true)] Type? c) { throw new NotSupportedException(); }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:DynamicallyAccessedMemberTypes",
+            Justification = "Interfaces used in Reflection.Emit either dynamic or accessible")]
+        public override bool IsAssignableFrom([NotNullWhen(true)] Type? c)
+        {
+            if (c == null)
+            {
+                return false;
+            }
+
+            if (this == c || IsAssignableFrom(c.BaseType))
+            {
+                return true;
+            }
+
+            if (!c.IsConstructedGenericType)
```

Comment:
This will return wrong result when non-generic interface inherits instantiation of a generic interface.
```
using System.Reflection;
using System.Reflection.Emit;

Console.WriteLine(typeof(I1<int>).IsAssignableFrom(typeof(I2)).ToString());

var ab = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("MyAssembly"), AssemblyBuilderAccess.Run);
var mb = ab.DefineDynamicModule("MyModule");

var i1_T = mb.DefineType("I1",  TypeAttributes.Interface | TypeAttributes.Abstract);
i1_T.DefineGenericParameters(new[] { "T" });

var i1_int = i1_T.MakeGenericType(typeof(int));

var i2 = mb.DefineType("I2", TypeAttributes.Interface | TypeAttributes.Abstract);
i2.AddInterfaceImplementation(i1_int);

Console.WriteLine(i1_int.IsAssignableFrom(i2).ToString());

interface I1<T>
{
}

interface I2 : I1<int>
{
}
```
Expected result: "true" "true"
Actual result: "true" "false"